### ISSUE TITLE
Creating a model raises a juju.errors.JujuAPIError exception

### ISSUE THREAD

**Issue by gnuoy** [juju/python-libjuju#333]
Creating a model with libjuju is raising an exception:

```
(func-travis) liam@figroll:~/branches/zaza$ juju --version
2.6.6-disco-amd64
(func-travis) liam@figroll:~/branches/zaza$ pip freeze | grep juju
juju==0.11.6
juju-wait==2.7.0
jujubundlelib==0.5.6
```
Trying to create a model results in:

```
(func-travis) liam@figroll:~/branches/zaza$ cat /tmp/test-controller.py
#!/usr/bin/env python3

from juju import loop
from juju.controller import Controller


async def add_model(model_name):
    controller = Controller()
    await controller.connect()
    model = await controller.add_model(model_name)
    await model.disconnect()
    await controller.disconnect()


loop.run(add_model('testmodel'))
(func-travis) liam@figroll:~/branches/zaza$ /tmp/test-controller.py
Error in watcher
Traceback (most recent call last):
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/model.py", line 814, in _all_watcher
    loop=self._connector.loop)
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/utils.py", line 113, in run_with_interrupt
    return task.result()  # may raise exception
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/client/facade.py", line 420, in wrapper
    reply = await f(*args, **kwargs)
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/client/_client1.py", line 161, in Next
    reply = await self.rpc(msg)
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/client/overrides.py", line 101, in rpc
    result = await client.WatchAll()
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/client/facade.py", line 420, in wrapper
    reply = await f(*args, **kwargs)
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/client/_client2.py", line 4207, in WatchAll
    reply = await self.rpc(msg)
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/client/facade.py", line 545, in rpc
    result = await self.connection.rpc(msg, encoder=TypeEncoder)
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/client/connection.py", line 335, in rpc
    raise errors.JujuAPIError(result)
juju.errors.JujuAPIError: model "8dced74e-50a4-466d-8bcb-40e72effe0a1" not found
Task exception was never retrieved
future: <Task finished coro=<Model._watch.<locals>._all_watcher() done, defined at /home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/model.py:805> exception=JujuAPIError('model "8dced74e-50a4-466d-8bcb-40e72effe0a1" not found')>
Traceback (most recent call last):
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/model.py", line 814, in _all_watcher
    loop=self._connector.loop)
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/utils.py", line 113, in run_with_interrupt
    return task.result()  # may raise exception
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/client/facade.py", line 420, in wrapper
    reply = await f(*args, **kwargs)
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/client/_client1.py", line 161, in Next
    reply = await self.rpc(msg)
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/client/overrides.py", line 101, in rpc
    result = await client.WatchAll()
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/client/facade.py", line 420, in wrapper
    reply = await f(*args, **kwargs)
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/client/_client2.py", line 4207, in WatchAll
    reply = await self.rpc(msg)
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/client/facade.py", line 545, in rpc
    result = await self.connection.rpc(msg, encoder=TypeEncoder)
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/client/connection.py", line 335, in rpc
    raise errors.JujuAPIError(result)
juju.errors.JujuAPIError: model "8dced74e-50a4-466d-8bcb-40e72effe0a1" not found
^CTraceback (most recent call last):
  File "/tmp/test-controller.py", line 15, in <module>
    loop.run(add_model('testmodel'))
  File "/home/liam/branches/zaza/.tox/func-travis/lib/python3.7/site-packages/juju/loop.py", line 36, in run
    raise KeyboardInterrupt()
KeyboardInterrupt
```

Interestingly the model is created (and the new models uuid matches the one in the stacktrace above):

```
$ juju show-model testmodel
testmodel:
  name: admin/testmodel
  short-name: testmodel
  model-uuid: 8dced74e-50a4-466d-8bcb-40e72effe0a1
  model-type: iaas
  controller-uuid: 74d25c5f-44ad-4110-895a-6edc35843d59
  controller-name: localhost-localhost
  is-controller: false
  owner: admin
  cloud: localhost
  region: localhost
  type: lxd
  life: alive
  status:
    current: available
    since: 1 minute ago
  users:
    admin:
      display-name: admin
      access: admin
      last-connection: 1 minute ago
  sla: unsupported
  agent-version: 2.6.6
  credential:
    name: localhost
    owner: admin
    cloud: localhost
```

**Comment by mitechie** [comment]
I think this is the race that was found/fixed in https://bugs.launchpad.net/juju/+bug/1838774

It would be good to see if this is repeatable and if a small delay after add-model lets the cache update and catch the new model and if that then passes on the current edge snap.

**Comment by mitechie** [comment]
Confirmed by putting a delay after the add-model call allowed it to work. This has a fix in 2.6 tree and we need to get it out to users.
